### PR TITLE
Add deposit limits for known tokens

### PR DESCRIPTION
**Pull request by elias-garcia**
### What does this PR does?

This PR adds two deposit limits for each known token (a `soft` limit and a `hard` limit). We will require user confirmation if the amount exceeds the soft limit and we'll directly forbid it if it exceeds the hard limit.

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Respect code style and lint
- [x] Update documentation (if needed)

### CODE REVIEW

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { SOFT_DEPOSIT_LIMIT_EXCEEDED_WORD } from "src/constants";
+
+import { FormData } from "src/domain";
+import { useDepositLimitModalStyles } from "src/views/home/components/deposit-limit-modal/deposit-limit-modal.styles";
+import { TextMatchForm } from "src/views/home/components/text-match-form/text-match-form.view";
+import { Button } from "src/views/shared/button/button.view";
+import { Card } from "src/views/shared/card/card.view";
+import { Portal } from "src/views/shared/portal/portal.view";
+
+export type DepositLimitModalData =
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:12]: I'd name this just `Limit` or `DepositLimit` and would use the same name for the `data` prop below.

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      formData: FormData;
+      type: "warning";
+    }
+  | {
+      formData: FormData;
+      hardLimit: number;
+      type: "forbidden";
+    };
+
+interface DepositLimitProps {
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: `DepositLimitModalProps`

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            <button className={classes.cancelButton} onClick={onCancel} type="button">
+              Cancel
+            </button>
+          </Card>
+        ) : (
+          <Card className={classes.card}>
+            <h1 className={classes.title}>Warning</h1>
+            <p className={classes.forbiddenText}>
+              For safety reasons we don&apos;t allow to deposit large quantities of tokens into the
+              zkEVM by the moment. Please, try again with an amount smaller than{" "}
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: I'd change this to:

```
For safety reasons, we do not yet allow deposits of large amounts of tokens into zkEVM. Please try again with an amount less than
```

File: src/views/home/components/text-match-form/text-match-form.view.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+interface TextMatchFormProps {
+  onSubmit: () => void;
+  text: string;
+}
+
+export const TextMatchForm: FC<TextMatchFormProps> = ({ onSubmit, text }) => {
+  const classes = useTextMatchFormStyles();
+  const [inputValue, setInputValue] = useState("");
+
+  const doTextsMatch = () => text.toLowerCase() === inputValue.toLowerCase();
```
Comment (amonsosanz) [inline src/views/home/components/text-match-form/text-match-form.view.tsx]: Why is this a function?

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import { FC } from "react";
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:1]: Don't you want to close this modal with ESC or a click in the background as we normally do?

File: src/views/home/home.view.tsx [hunk 6]
```diff
@@ -17,12 +24,37 @@ export const Home = (): JSX.Element => {
...
-  const onFormSubmit = (formData: FormData) => {
+  const onSubmitForm = (formData: FormData) => {
     setFormData(formData);
     navigate(routes.bridgeConfirmation.path);
   };
 
+  const onCheckDepositLimitAndSubmitForm = (formData: FormData) => {
+    if (formData.from.key === "ethereum") {
+      const { from, token } = formData;
+      const depositLimits = DEPOSIT_LIMITS[from.chainId][token.address];
```
Comment (amonsosanz) [inline src/views/home/home.view.tsx:39]: This is `undefined` for any token not present in `DEPOSIT_LIMITS` and raises runtime errors below.

image

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 5 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:1]: Nope, in this case I prefer to avoid that users accidentally click outside of the modal to close it

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 1 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:12]: I've already created a `DepositLimit` interface in the domain:

```ts
interface DepositLimit {
  hard: number;
  soft: number;
}
```

Take into account that this interface refers to the Modal being shown `warning` or `forbidden`, I think that the current name really describes what it is.

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 3 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: The text is going to be provided by the legal team, but I've changed it just in case they don't provide it 🤣

File: src/views/home/home.view.tsx (hunk 6 above)
Comment (elias-garcia) [reply src/views/home/home.view.tsx:39]: Right! Good catch, let me fix it.

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+              );
+            }
+            case "unknown-limit": {
+              return (
+                <Card className={classes.card}>
+                  <Typography className={classes.title} type="h1">
+                    Warning
+                  </Typography>
+                  <Typography className={classes.warningText} type="body1">
+                    We do not oficcialy support the token you are about to bridge. For safety
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: There's a typo in `oficcialy`. I'd rephrase it as:

```
We do not officially support the token you are about to bridge. For safety reasons, we must ensure you understand the risks and the possible consequences of something going wrong.
```

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 3 above)
Comment (amonsosanz) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: Sorry I just realized I missed an article:

```
For safety reasons, we do not yet allow deposits of large amounts of tokens into the zkEVM. Please try again with an amount less than
```

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    }
+  | {
+      formData: FormData;
+      hardLimit: number;
+      type: "forbidden";
+    }
+  | {
+      formData: FormData;
+      type: "unknown-limit";
+    };
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:25]: It's not that it's very relevant but since `formData: FormData` is present in the 3 cases, I'd move it to another prop to reduce verbosity:

```typescript
export type DepositLimitModalData =
  | {
      type: "warning";
    }
  | {
      hardLimit: number;
      type: "forbidden";
    }
  | {
      type: "unknown-limit";
    };

interface DepositLimitModalProps {
  data: DepositLimitModalData;
  formData: FormData;
  onAccept: (formData: FormData) => void;
  onCancel: () => void;
}
```

**Review by amonsosanz — APPROVED** [review]

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 8 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx:25]: There is a problem if you do it like that hehe. As the modal check happens before setting the `formData` I don't have it available, so it's easier to pass it down like this.

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 7 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: Right! Fixed

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx (hunk 3 above)
Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: Updated!

File: src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx [hunk 9]
```diff
@@ -88,9 +88,9 @@ export const DepositLimitModal: FC<DepositLimitModalProps> = ({ data, onAccept,
                     Warning
                   </Typography>
                   <Typography className={classes.forbiddenText} type="body1">
-                    We do not officially support the token you are about to bridge. For safety
-                    reasons, we need to ensure that you understand the risks and the possible
-                    consequences of something going wrong.
+                    You are about to deposit an imported token into the Mainnet Beta of the zkEVM.
+                    For safety reasons, we need to ensure that you are understand its risks and the
```
Comment (amonsosanz) [inline src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: `that you understand`

Comment (elias-garcia) [reply src/views/home/components/deposit-limit-modal/deposit-limit-modal.view.tsx]: Changed!
